package/MDAnalysis/converters/OpenMMParser.py

```diff
@@ -184,26 +181,39 @@ def _mda_topology_from_omm_topology(self, omm_topology):
                 warnings.warn("Element information missing for some atoms. "
                               "These have been given an empty element record ")
                 if any(i == 'X' for i in atomtypes):
-                    warnings.warn("For absent elements, atomtype has been  "
-                                  "set to 'X' and mass has been set to 0.0. "
-                                  "If needed these can be guessed using "
-                                  "MDAnalysis.topology.guessers.")
+                    warnings.warn(
+                        "For absent elements, atomtype has been  "
+                        "set to 'X' and mass has been set to 0.0. "
+                        "If needed these can be guessed using "
+                        "universe.guess_TopologyAttrs("
+                        "to_guess=['masses', 'types']). "
+                        "(for MDAnalysis version 2.x "
+                        "this is done automatically,"
+                        " but it will be removed in future versions).")
+
                 attrs.append(Elements(np.array(validated_elements,
                                                dtype=object)))
+                attrs.append(Atomtypes(np.array(atomtypes, dtype=object)))
+                attrs.append(Masses(np.array(masses)))
 
             else:
-                atomtypes = guess_types(atomnames)
-                masses = guess_masses(atomtypes)
-                wmsg = ("Element information is missing for all the atoms. "
-                        "Elements attribute will not be populated. "
-                        "Atomtype attribute will be guessed using atom "
-                        "name and mass will be guessed using atomtype."
-                        "See MDAnalysis.topology.guessers.")
+                wmsg = (
+                    "Element information is missing for all the atoms. "
+                    "Elements attribute will not be populated. "
+                    "Atomtype attribute will be guessed using atom "
+                    "name and mass will be guessed using atomtype."
+                    "for MDAnalysis version 2.x this is done automatically, "
+                    "but it will be removed in future versions. "
+                    "These can be guessed using "
+                    "universe.guess_TopologyAttrs("
+                    "to_guess=['masses', 'types']) "
+                    "See MDAnalysis.guessers.")
+
                 warnings.warn(wmsg)
         else:
             attrs.append(Elements(np.array(validated_elements, dtype=object)))
-        attrs.append(Atomtypes(np.array(atomtypes, dtype=object)))
-        attrs.append(Masses(np.array(masses)))
+            attrs.append(Atomtypes(np.array(atomtypes, dtype=object)))
+            attrs.append(Masses(np.array(masses)))
```

Comment:
Is it correct that these lines got added into the `else` block? The fact that they are not covered by tests at least makes me ask.

Reply:
I rearranged them in an old commit for a behavior that does not exist anymore.. so I'll remove it as it has no value